JIP-57 권한 확인 middleware

tmam444 opened the pull request:
### 개요
권한 확인 middleware 추가

### 작업 사항
- auth.interface.ts에 Role과 RoleSet 추가
- auth.validate.ts에 권한 확인할 middleware 추가
- auth.jwt.ts에 jwt token payload 수정 (role 삭제)

### 변경점
- 권한 확인 middleware 추가

### 목적
- 각 API당 권한을 확인하고 작업하기 위해

### 스크린샷 (optional)

min-kyung reviewed (approved):
저번에 말씀하신 fixture를 사용하여 데이터를 임시 저장하는 코드 확인했습니다.

tmam444 committed:
JIP-57 권한 확인 middleware
    - auth.interface.ts에 Role과 RoleSet 추가
    - auth.validate.ts에 권한 확인할 middleware 추가
    - auth.jwt.ts에 jwt token payload 수정 (role 삭제)

tmam444 committed:
JIP-57 Auth Pull request 전 수정
    - interface.ts를 type.ts로 변경
    - CamelCase로 enum 명 설정

kth2624 reviewed (approved):
변경사항 확인했습니다
